dnfdaemon-server/main.cpp

```diff
@@ -24,10 +24,25 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include <sdbus-c++/sdbus-c++.h>
 
 #include <iostream>
+#include <signal.h>
 
 static std::unique_ptr<SessionManager> session_manager = nullptr;
 
+void sigint_handler(int signum) {
+    std::cout << "received signal " << signum << std::endl;
+    session_manager->shut_down();
+    if (session_manager != nullptr) {
+        session_manager->shut_down();
+    }
+}
+
 int main() {
+    struct sigaction SignalAction;
```

Comment:
I think `SignalAction` should be `signal_action` to adhere to our naming convetion, it's a variable.